### DISCUSSION TITLE
How to run scalene inside Autodesk Maya or any other DCC

### DISCUSSION THREAD

**Discussion by rchals** [plasma-umass/scalene#505] [Q&A] [answered]
Hello,
I'm trying to run scalene inside Autodesk Maya 2023 from the script editor, but I get this warning:
![image](https://user-images.githubusercontent.com/5830049/206473125-77a7472f-29fd-4eec-8e1f-65228be219e9.png)

I've seen other answers about this issue, but they rely on launching scalene like `scalene foo.py`, and I'm not quite sure how I can workaround this from inside the live script editor.

For context, I used to perform profiling using cProfile inside maya using this:
![image](https://user-images.githubusercontent.com/5830049/206473797-323c8989-bae4-4b27-9b12-f2a52365e74b.png)

Any ideas?
Thanks!

**Comment by emeryberger** [comment]
Can you run code by running a module first? (That is, invoking it as `python3 -m scalene ...`?)

**Reply by rchals** [reply]
Thanks for the quick response!

I cannot launch command line python inside Maya, it needs to be python code.
However, I tried to run the module using runpy but I get the same error.

```
import runpy
d = runpy.run_module('scalene')

def foo():
    from maya import cmds
    for _ in range(50):
        cmds.createNode('locator')

# Turn profiling on
d['scalene_profiler'].start()

foo()

# Turn profiling off
d['scalene_profiler'].stop()
```

_edit: Tried to use the internal variables given by the runpy, the "start" error keeps appearing_

**Comment by emeryberger** [comment, marked as answer]
Scalene pre-loads some libraries that enable its memory profiling and other features. If you don't need those and are only interested in CPU/GPU profiling, it should be possible to run Scalene programmatically without invoking Scalene first, but this is not something we currently support.

**Reply by rchals** [reply]
Thank you very much, I will look foward for new releases. Keep up the good work!

**Reply by emeryberger** [reply]
If you don't mind, could you re-post this as a feature request issue? That way, we won't lose track of it. Thanks (i pel teu nom, merci!)!

**Reply by rchals** [reply]
Sure thing, cap problema :-), I'll do it right now. Moltes gràcies!

**Comment by rafaelperez** [comment]
@rchals @emeryberger I'm in the same situation, where I'd like to run inside Foundry's Nuke - basically the 2D version of Autodesk Maya. 

I cannot upgrade Scalene from 1.5.4 since Nuke uses Python 3.7. However, I would be OK profiling just the CPU.

What are the steps for doing that?

**Reply by rafaelperez** [reply]
I've tried the following so far:

`scalene_profiler.start()`
and
`scalene_profiler.Scalene.main()`

Both commands close Nuke.

**Reply by emeryberger** [reply]
You still need to somehow invoke the code with Scalene (e.g., via `scalene` or `python3 -m scalene`).
